sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/healthcare/DicomIOReadIT.java

```diff
@@ -90,4 +94,29 @@ public void testDicomMetadataRead() throws IOException {
       // noop - if runner does not support job.cancel()
     }
   }
+
+  @Test
+  public void testDicomFailedMetadataRead() throws IOException {
+    String badWebPath = "foo";
+
+    DicomIO.ReadStudyMetadata.Result result =
+        pipeline.apply(Create.of(badWebPath)).apply(DicomIO.readStudyMetadata());
+
+    PAssert.that(result.getReadResponse()).empty();
+
+    PAssert.that(result.getFailedReads())
+        .satisfies(
+            (errors) -> {
+              Assert.assertTrue(errors.iterator().hasNext());
+              return null;
+            });
+
+    PipelineResult job = pipeline.run();
+
+    try {
+      job.cancel();
```

Comment:
I was confused here in both tests. `job.cancel()` immediately after pipeline.run(). pipeline is a TestPipeline so `pipeline.run()` is a blocking call until the pipeline finished. What was the purpose of having `job.cancel()` here?

Reply:
Thanks, you're right in that its not needed